clients/src/main/java/org/apache/kafka/common/errors/ResourceNotFoundException.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * Exception thrown due to a request for a resource that does not exist.
+ */
+public class ResourceNotFoundException extends ApiException {
+    private final String resource;
+
```

Comment:
We've always set `serialVersionUID` in exception classes, to appease the Java gods.
```
    private static final long serialVersionUID = 1L;
```
it's not really necessary since we don't plan on using Java's built-in serialization framework, but I think some tools complain if they don't see it.  And anyway, it's the pattern... for now.